fingerprint/capture/src/main/java/com/simprints/fingerprint/capture/screen/FingerprintCaptureFragment.kt

```diff
@@ -164,14 +168,34 @@ internal class FingerprintCaptureFragment : Fragment(R.layout.fragment_fingerpri
             binding.fingerprintViewPager,
             binding.fingerprintIndicator,
             onFingerSelected = { position -> vm.updateSelectedFinger(position) },
-            isAbleToSelectNewFinger = { vm.stateLiveData.value?.currentCaptureState()?.isCommunicating() != true }
+            isAbleToSelectNewFinger = { vm.stateLiveData.value?.currentCaptureState()?.isCommunicating() != true },
+            onPageScrolled = { position: Int, positionOffset: Float ->
+                if(positionOffset != 0.0f){
+                    vm.stateLiveData.value?.fingerStates?.let { fingerStates ->
+                        val nextPage = if (positionOffset > 0.0f) {
+                            position + 1
+                        } else {
+                            position
+                        }
+                        val currentColor = fingerStates[position].currentCapture().statusBarColor()
+                        val nextColor = fingerStates[nextPage].currentCapture().statusBarColor()
+                        if (currentColor != nextColor) {
+                            val color =ColorUtils.blendARGB(
+                                ContextCompat.getColor(requireContext(), currentColor),
```

Comment:
Could this be cached in some way? From the look of it, this could be a pretty hot path, and resource lookups are usually fairly expensive.

Reply:
Even though I don't think it takes _that_ much resources to animate the color change, it can be cached, yes. I will take a look at it once I'm back

Reply:
Cache is added and is ready for review @luhmirin-s